tests/Feature/Facades/SpApiFacadeTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Tests\Feature\Facades;
+
+use Glue\SpApi\Laravel\Facades\SpApi;
+use Tests\TestCase;
+
+class SpApiFacadeTest extends TestCase
+{
+    public function test_it_instantiates_new_facade_roots_on_each_static_call()
+    {
+        $spApi1 = SpApi::getFacadeRoot();
+        $spApi2 = SpApi::getFacadeRoot();
+
+        $this->assertNotSame($spApi1, $spApi2);
+    }
+
```

Comment:
But here since you do have the `use RefreshesFacadeInstance;` trait on the SpApi facade class it should create a new instance on each static call of `getFacadeRoot()`?

Reply:
Bingo - and by extension any static call to the `SpApi` facade, since under the hood it calls `getFacadeRoot` to resolve the instance.

Reply:
Ohh okay so that `getFacadeRoot` method is always called first even when other methods are called. So as you said it'll always be a new instance when calling any static method.